base/broadcast.jl

```diff
@@ -4,19 +4,30 @@ module Broadcast
 
 using Base.Cartesian
 using Base: promote_eltype_op, linearindices, tail, OneTo, to_shape,
-            _msk_end, unsafe_bitgetindex, bitcache_chunks, bitcache_size, dumpbitcache
+            _msk_end, unsafe_bitgetindex, bitcache_chunks, bitcache_size, dumpbitcache,
+            promote_op, nullable_returntype, null_safe_eltype_op, hasvalue,
+            is_nullable_array
 import Base: .+, .-, .*, ./, .\, .//, .==, .<, .!=, .<=, .÷, .%, .<<, .>>, .^
 import Base: broadcast
 export broadcast!, bitbroadcast, dotview
 export broadcast_getindex, broadcast_setindex!
 
 ## Broadcasting utilities ##
 
+broadcast_array_type() = Array
```

Comment:
It's used as a marker to select whether to recurse or otherwise. Perhaps that's best done with a different trait.